silk/datadog_checks/silk/check.py

```
@@ -0,0 +1,224 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from copy import deepcopy
+
+from six.moves.urllib.parse import urljoin, urlparse
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.utils.time import get_timestamp
+
+from .events import SilkEvent
+from .metrics import BLOCKSIZE_METRICS, METRICS, READ_WRITE_METRICS
+
+EVENT_PATH = "events?timestamp__gte={}"
+
+STATE_ENDPOINT = 'system/state'
+SERVERS_ENDPOINT = 'system/servers'
+
+STATE_MAP = {'online': AgentCheck.OK, 'offline': AgentCheck.WARNING, 'degraded': AgentCheck.CRITICAL}
+
+
+class SilkCheck(AgentCheck):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'silk'
+
+    STATE_SERVICE_CHECK = "system.state"
+    SERVERS_SERVICE_CHECK = "server.state"
+    CONNECT_SERVICE_CHECK = "can_connect"
+    METRICS_TO_COLLECT = METRICS
+
+    def __init__(self, name, init_config, instances):
+        super(SilkCheck, self).__init__(name, init_config, instances)
+
+        server = self.instance.get("host_address")
```

Can we also raise an error here when `host_address` is not provided?